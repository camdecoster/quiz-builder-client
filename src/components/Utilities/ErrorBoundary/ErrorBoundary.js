import React, { Component } from "react";
import PropTypes from "prop-types";
import "./ErrorBoundary.css";

export default class ErrorBoundary extends Component {
    static defaultProps = {
        message: `Something didn't load correctly`,
    };

    constructor(props) {
        super(props);
        this.state = {
            hasError: false,
        };
    }

    static getDerivedStateFromError(error) {
        return { hasError: true };
    }

    render() {
        if (this.state.hasError) {
            return (
                <h3 className='errorMessage'>
                    There was an error: {this.props.message}.
                </h3>
            );
        }
        return this.props.children;
    }
}

ErrorBoundary.propTypes = {
    message: PropTypes.string.isRequired,
};
