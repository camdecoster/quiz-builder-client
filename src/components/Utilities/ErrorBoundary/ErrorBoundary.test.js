// React
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

// Configuration
// import QuizBuilderContext from "../../contexts/QuizBuilderContext";
// import TokenService from "../../services/token-service";

// Components
import ErrorBoundary from "./ErrorBoundary";

function ComponentWithError() {
    throw new Error("Error caused by test component");
}

describe("<ErrorBoundary> component", () => {
    it("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(
            <ErrorBoundary>
                <ComponentWithError />
            </ErrorBoundary>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    describe("renders the UI", () => {
        it("with the default message (no message given)", () => {
            const tree = renderer
                .create(
                    <ErrorBoundary>
                        <ComponentWithError />
                    </ErrorBoundary>
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it("with the given message", () => {
            const tree = renderer
                .create(
                    <ErrorBoundary message='This is an error message'>
                        <ComponentWithError />
                    </ErrorBoundary>
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
