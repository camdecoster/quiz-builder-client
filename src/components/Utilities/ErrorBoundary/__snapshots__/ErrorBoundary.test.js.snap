// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary> component renders the UI with the default message (no message given) 1`] = `
<h3
  className="errorMessage"
>
  There was an error: 
  Something didn't load correctly
  .
</h3>
`;

exports[`<ErrorBoundary> component renders the UI with the given message 1`] = `
<h3
  className="errorMessage"
>
  There was an error: 
  This is an error message
  .
</h3>
`;
